src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx

```diff
@@ -34,61 +32,24 @@ export type PupilViewsProgrammeListingProps = {
   baseSlug: string;
   yearSlug: PupilProgrammeListingData["yearSlug"];
   examboardSlug?: ExamboardData["examboardSlug"];
+  examboards: ExamboardData[];
+  tiers: TierData[];
 };
 
 export const PupilViewsProgrammeListing = ({
   programmes,
   baseSlug,
   yearSlug,
   examboardSlug,
+  examboards,
+  tiers,
 }: PupilViewsProgrammeListingProps) => {
-  const allExamboards: { [key: string]: ExamboardData[] } = _.groupBy(
-    getAvailableProgrammeFactor({
-      programmes,
-      factorPrefix: "examboard",
-    }) as ExamboardData[],
-    (examboard: ExamboardData) => examboard.examboard,
-  );
-
-  // This creates an array of examboards giving preference to non-legacy examboards
-  const examboards = Object.keys(allExamboards)
-    .map((examboard) => {
-      const mappedExamboard = allExamboards[examboard];
-      if (!Array.isArray(mappedExamboard) || mappedExamboard.length < 1) return;
-      return (
-        allExamboards[examboard]?.find(
-          (examboard: ExamboardData) => !examboard.isLegacy,
-        ) ?? mappedExamboard[0]
-      );
-    })
-    .filter((examboard): examboard is ExamboardData => examboard !== undefined);
-
   const [chosenExamboard, setChosenExamboard] = useState<ExamboardData | null>(
     examboardSlug && examboards.length >= 1
       ? getExamboardData({ examboardSlug, availableExamboards: examboards })
       : null,
   );
 
-  const allTiers: { [key: string]: TierData[] } = _.groupBy(
-    getAvailableProgrammeFactor({
-      programmes: programmes,
-      factorPrefix: "tier",
-    }) as TierData[],
-    (tier: TierData) => tier.tier,
-  );
-
-  // This creates an array of tiers giving preference to non-legacy tiers
-  const tiers = Object.keys(allTiers)
-    .map((tierLabel) => {
-      const mappedTier = allTiers[tierLabel];
-      if (!Array.isArray(mappedTier) || mappedTier.length < 1) return;
-      return (
-        allTiers[tierLabel]?.find((tier: TierData) => !tier.isLegacy) ??
-        mappedTier[0]
-      );
-    })
-    .filter((tier): tier is TierData => tier !== undefined);
```

Comment:
Moved to the helper function that is called in getStaticProps - options-pages-helpers.ts